storybookjs/storybook
# Make `transform` Source API consistent

Closes #21744



## What I did

This PR makes the `transformSource` parameter more consistent with the rest of the docs API:

- Renamed to `parameters.docs.source.transform`
- supported as a prop on `<Source transform={(src, story) => {...}} />` and `<Canvas source={{ transform: (src, story) => {...} }} />`
- Removed transformations happening directly in the source decorators in `html`, `web-components` and `react`, to make them happen in the `Source` block instead.



## How to test

See stories: (waiting for CI to finish, will attach links)
- `Source` block
- template story (scroll to bottom, Show Code at last story)
- Internal stories for deprecated parameters.

@kylegach could you review the docs changes? https://github.com/storybookjs/storybook/pull/21749/files#diff-c469f1732b859e1892b7d70dc37b4de48c7de12c6c6b8dfb72b4d3a5d7af2c79



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
